framework/test/unit/modules/random/endpoint.spec.ts

```diff
@@ -255,6 +295,123 @@ describe('RandomModuleEndpoint', () => {
 			);
 		});
 
+		it('should throw error when distance is greater than count', async () => {
+			// Arrange
+			const { address } = genesisValidators.validators[0];
+			const seed = '7c73f00f64fcebbab49a6145ef14a843';
+			const count = 1000;
+			const distance = 1001;
+			context.params = { address, seed, count, distance };
+
+			// Act & Assert
+			await expect(randomEndpoint.setHashOnion(context)).rejects.toThrow(
+				'Invalid distance. Distance must be less than or equal to count',
+			);
+		});
+
+		it('should throw error when hashes is provided but not count', async () => {
+			// Arrange
+			const { address } = genesisValidators.validators[0];
+			const distance = 1001;
+			context.params = { address, hashes: ['7c73f00f64fcebbab49a6145ef14a843'], distance };
+
+			// Act & Assert
+			await expect(randomEndpoint.setHashOnion(context)).rejects.toThrow(
+				'Hashes must be provided with count and distance.',
+			);
+		});
+
+		it('should throw error when hashes is provided but not distance', async () => {
+			// Arrange
+			const { address } = genesisValidators.validators[0];
+			const count = 1000000;
+			context.params = { address, hashes: ['7c73f00f64fcebbab49a6145ef14a843'], count };
+
+			// Act & Assert
+			await expect(randomEndpoint.setHashOnion(context)).rejects.toThrow(
+				'Hashes must be provided with count and distance.',
+			);
+		});
+
+		it('should throw error when hashes empty', async () => {
```

Comment:
```suggestion
		it('should throw error when hashes has an element not 16 bytes', async () => {
```

Reply:
rephrased the error message. the test case was switched with something else, and `empty` was correct to check the empty array